src/snapshot.rs

```diff
@@ -0,0 +1,57 @@
+#![allow(dead_code)]
+use std::{fs, io};
+
+use ethers::types::U256;
+use indexmap::IndexSet;
+use serde::{Deserialize, Serialize};
+
+use crate::GENESIS_BLOCK;
+
+/// Struct containing the fields used for restoring the tree state.
+#[derive(Serialize, Deserialize)]
+pub struct StateSnapshot {
+    /// The latest l1 block number that was processed.
+    pub current_l1_block_number: u64,
```

Comment:
I just realized something: When we import state from other sources (e.g. file), there's no such thing as L1 block number, right?

Reply:
That might be the case, I'm not sure what the file format would look like at this point so I can't really answer that. Either way though, I think we can defer this issue to a later PR, what do you say?

Reply:
Yes, definitely. I'm actually working on async `L1Fetcher` right now, based on your code here, so hence the thought. Let's see how this gets shape :slightly_smiling_face: